lib/search/ui/widgets/search_bar.dart

```diff
@@ -0,0 +1,190 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:fluttercon/common/utils/router.dart';
+import 'package:fluttercon/common/widgets/resolved_image.dart';
+import 'package:fluttercon/core/di/injectable.dart';
+import 'package:fluttercon/core/theme/theme_colors.dart';
+import 'package:fluttercon/search/cubit/search_cubit.dart';
+import 'package:fluttercon/search/cubit/search_state.dart';
+import 'package:fluttercon/search/models/search_result.dart';
+import 'package:go_router/go_router.dart';
+
+class SearchBarWidget extends StatelessWidget {
+  const SearchBarWidget({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => getIt<SearchCubit>(),
+      child: const SearchBarView(),
+    );
+  }
+}
+
+class SearchBarView extends StatefulWidget {
+  const SearchBarView({Key? key}) : super(key: key);
+
+  @override
+  _SearchBarViewState createState() => _SearchBarViewState();
+}
+
+class _SearchBarViewState extends State<SearchBarView> {
+  final TextEditingController _searchController = TextEditingController();
+  Timer? _debounce;
+
+  @override
+  void dispose() {
+    _debounce?.cancel();
+    _searchController.dispose();
+    super.dispose();
+  }
+
+  void _onSearchChanged(String query) {
+    if (_debounce?.isActive ?? false) _debounce?.cancel();
+    _debounce = Timer(const Duration(milliseconds: 300), () {
+      if (query.isNotEmpty) {
+        context.read<SearchCubit>().search(query);
+      } else {
+        context.read<SearchCubit>().clearSearch();
+      }
+    });
+  }
+
+  void _clearSearch() {
+    _searchController.clear();
+    FocusScope.of(context).unfocus();
+    context.read<SearchCubit>().clearSearch();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Container(
+          padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
+          decoration: BoxDecoration(
+            color: Theme.of(context).cardColor,
+            borderRadius: BorderRadius.circular(35),
+            border: Border.all(color: ThemeColors.blueDroidconColor),
+          ),
+          child: Row(
+            children: [
+              const Icon(Icons.search,
+                  color: ThemeColors.blueGreenDroidconColor),
+              const SizedBox(width: 8),
+              Expanded(
+                child: TextField(
+                  controller: _searchController,
+                  decoration: InputDecoration(
+                    hintText: 'Search sessions or speakers',
```

Comment:
Move this to the `.arb` file. Steps are available on another comment.

Reply:
done